index.js

```diff
@@ -939,26 +938,27 @@
 
     return isEmpty(values) ?
       [Unknown] :
-      or(Z.reduce(refine, env, values), loose ? [Inconsistent] : []);
+      or(Z.reduce(refine, env, values), [Inconsistent]);
   }
 
-  //  rejectInconsistent :: Array Type -> Array Type
-  function rejectInconsistent(types) {
-    return types.filter(function(t) {
-      return t.type !== INCONSISTENT && t.type !== UNKNOWN;
-    });
+  //  isConsistent :: Type -> Boolean
+  function isConsistent(t) {
+    return t.type === UNARY   ? isConsistent(t.types.$1.type) :
+           t.type === BINARY  ? isConsistent(t.types.$1.type) &&
+                                isConsistent(t.types.$2.type) :
+           /* else */           t.type !== INCONSISTENT;
   }
```

Comment:
This function is an island of sanity in a sea of complicated code. I'm pleased that we're no longer treating `$.Unknown` as inconsistent. :)